drivers/pipes/pipe_common.c

```diff
@@ -105,6 +105,8 @@ FAR struct pipe_dev_s *pipecommon_allocdev(size_t bufsize)
       nxmutex_init(&dev->d_bflock);
       nxsem_init(&dev->d_rdsem, 0, 0);
       nxsem_init(&dev->d_wrsem, 0, 0);
+      nxsem_init(&dev->d_nrdsem, 0, 0);
+      nxsem_init(&dev->d_nwrsem, 0, 0);
```

Comment:
why not to reuse d_wrsem?

Reply:
`d_wrsem` (and `d_rdsem`) could both be reused (in fact, `d_rdsem` was being reused prior this PR). However, these semaphores had well-defined meanings:
```
  sem_t      d_rdsem;       /* Empty buffer - Reader waits for data write */
  sem_t      d_wrsem;       /* Full buffer - Writer waits for data read */
```

I preferred not to reuse them following the _explicit is better than implicit_. Let me know if you prefer to save some bytes instead and I'll change it ;)

Reply:
since the old design reuse rdsem, I would prefer that we follow the original design to minimize the code change. If you think the separation, make the code clean, it's better to do the refactor in a new patch either before or after this change.